### ISSUE TITLE
read failed, socket might closed or timeout, read ret: -1

### ISSUE THREAD

**Issue by angeldev** [mazenrashed/Printooth#44]
Hi, I have a list of items with a button which I can press and print the item info. Printing is fine if I press a button and wait for a while until I press again another one. But if I press two buttons too close in time I get this error and it prints only the first one:

read failed, socket might closed or timeout, read ret: -1

Any workaround I can have here?
Thanks!

**Comment by RaRoPe** [comment]
Yes, @angeldev , i got this error too. You solved it? I'm thinking that is an error in socket, that if we want to create a two instances of printing, we should create a different socket to support that. Or we could change the Bluetooth.java, but we can't do that because that file is protected. Another solution is trying to create our library changing the 

`device.createRfcommSocketToServiceRecord(mMyUuid);`

from:

`tmp = createBluetoothSocket(mmDevice);`

This one is the part of the code that we got our error:

(android/platform/frameworks/base/d6883533e4ac3f73d2fde1db9a1dddf06dac6709/./core/java/android/bluetooth/BluetoothSocket.java)

```
private int readAll(InputStream is, byte[] b) throws IOException {

   int left = b.length;
   while(left > 0) {
   int ret = is.read(b, b.length - left, left);
   if(ret <= 0)
   throw new IOException("read failed, socket might closed, read ret: " + ret);
   left -= ret;
   if(left != 0)
   Log.w(TAG, "readAll() looping, read partial size: " + (b.length - left) +
   ", expect size: " + b.length);
   }
   return b.length;
}
```

**Comment by filutfi** [comment]
hi @angeldev , did you solve this problem?

**Comment by rafahsborges** [comment]
I've changed the code in Bluetooth.java as follows:

ConnectThread(BluetoothDevice device, boolean insecureConnection) {
            Bluetooth.this.device = device;
            try {
                if (insecureConnection) {
                    Bluetooth.this.socket = device.createInsecureRfcommSocketToServiceRecord(uuid);
                } else {
                    Bluetooth.this.socket = (BluetoothSocket) device.getClass().getMethod("createRfcommSocket", new Class[] {int.class}).invoke(device,1);
                }
            } catch (IOException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
                if (deviceCallback != null) {
                    deviceCallback.onError(Objects.requireNonNull(e.getMessage()));
                }
            }
        }

It worked fine.

**Comment by grrigore** [comment]
any updates on this issue?
